### PR TITLE
LPD-15377 Normalize form rule expression to make sure the option label will be used instead of the option value

### PR DESCRIPTION
**Pull request by carolmariaabb**
https://liferay.atlassian.net/browse/LPD-15377

### CODE REVIEW

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/java/com/liferay/dynamic/data/mapping/form/builder/internal/converter/visitor/ConditionExpressionVisitor.java [hunk 1]
```diff
@@ -103,8 +104,14 @@ public Object visit(FunctionCallExpression functionCallExpression) {
...
 
+			String operandType = operand.getType();
+
+			if (StringUtil.equals(operand.getType(), "string")) {
+				operandType = "option";
+			}
+
 			return new SPIDDMFormRuleCondition.Operand(
-				"option", operand.getValue());
+				operandType, operand.getValue());
```
Comment (marcelabc) [inline modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/java/com/liferay/dynamic/data/mapping/form/builder/internal/converter/visitor/ConditionExpressionVisitor.java:114]: Curiosity: Is there other operand type besides "option" ?
I'm curious why we need to edit the way it was in the code ("option" hard-coded).

**Comment by marcelabc** [comment]
ci:test:relevant

**Comment by marcelabc** [comment]
ci:test:forms

**Comment by carolmariaabb** [comment]
ci:test:forms

File: modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/java/com/liferay/dynamic/data/mapping/form/builder/internal/converter/visitor/ConditionExpressionVisitor.java (hunk 1 above)
Comment (carolmariaabb) [reply modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-builder/src/main/java/com/liferay/dynamic/data/mapping/form/builder/internal/converter/visitor/ConditionExpressionVisitor.java:114]: These types are defined in [this](https://github.com/liferay/liferay-portal/blob/d60700a887fb5d7c57e142b4464eda04a4bef526/modules/apps/data-engine/data-engine-taglib/src/main/resources/META-INF/resources/data_layout_builder/js/components/rules/editor/config.es.js#L14-L20) FE code (hard-coded)

**Comment by carolmariaabb** [comment]
ci:test:sf

**Comment by carolmariaabb** [comment]
ci:test:forms

**Comment by carolmariaabb** [comment]
ci:forward:force

**Pull request closed without merging**